### PR TITLE
feat(main)!: only launch as root

### LINKED ISSUE #454: Launch `pacstall` using root
We should run the `pacstall` command using root, as most of the operations require root privileges.

We could later downscale privileges.

### PR DESCRIPTION
**Pull request by wizard-28**
BREAKING CHANGE: The program can only be launched as root.

## Purpose

Closes #454 

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

**Review by Henryws — APPROVED** [review]

File: pacstall/__main__.py [hunk 1]
```diff
@@ -41,16 +40,15 @@
...
-    if getuser() == "root":
-        message.fancy("error", "Pacstall cannot be run as root")
-        sys.exit(1)
-
     args = parse_arguments()
     print(args)
 
+    if getuser() != "root":
+        message.fancy("error", "Pacstall needs to be launched as root!")
+        sys.exit(64)  # --> command line usage error
```
Comment (paulcosma97) [inline pacstall/__main__.py:48]: If we're going to be using a set of standard error codes, we should either stick to the Linux Standard or create our own and reference them by a common enum

According to [this article](https://www.baeldung.com/linux/status-codes) code `126` would be better.

Comment (paulcosma97) [reply pacstall/__main__.py:48]: IMO we should have our own values hardcoded in some kind of enum and use that

Comment (wizard-28) [reply pacstall/__main__.py:48]: > If we're going to be using a set of standard error codes, we should either stick to the Linux Standard or create our own and reference them by a common enum
> 
> According to [this article](https://www.baeldung.com/linux/status-codes) code `126` would be better.

I got `64` from the [`sysexits.h`](https://www.freebsd.org/cgi/man.cgi?query=sysexits) (this is a standard UNIX header). From what I can see that article (it's exit codes) is for bash scripts (https://stackoverflow.com/a/1101969).

Comment (wizard-28) [reply pacstall/__main__.py:48]: > IMO we should have our own values hardcoded in some kind of enum and use that

Yeah, we could use that for specific error codes, but I think for common errors like this one we should follow the standard error codes.

Do you want me to add an Enum of all the error codes (the standard ones for now) in `pacstall/api/error_codes.py`?

Comment (paulcosma97) [reply pacstall/__main__.py:48]: Yup, that sounds good

Comment (wizard-28) [reply pacstall/__main__.py]: @paulcosma97 done

**Review by paulcosma97 — APPROVED** [review]

**Pull request merged**